src/main/java/com/conveyal/r5/streets/StreetRouter.java

```diff
@@ -228,8 +228,8 @@ public boolean setOrigin (double lat, double lon) {
         queue.clear();
         // from vertex is at end of back edge. Set edge correctly so that turn restrictions/costs are applied correctly
         // at the origin.
-        State startState0 = new State(split.vertex0, split.edge + 1, streetMode);
-        State startState1 = new State(split.vertex1, split.edge, streetMode);
+        State startState0 = new State(split.vertex0, -1, streetMode);
+        State startState1 = new State(split.vertex1, -1, streetMode);
```

Comment:
I'm not sure what the problem here was. I think we do want to set the backEdge on these states to be split.edge, so that turn restrictions are applied when leaving the edge that the origin was snapped to. Is there a reason this behavior was problematic?